Sources/AsyncHTTPClient/ConnectionsState.swift

```diff
@@ -15,29 +15,29 @@
 import NIO
 
 extension HTTP1ConnectionProvider {
-    enum Action {
-        case lease(Connection, Waiter)
-        case create(Waiter)
-        case replace(Connection, Waiter)
+    enum Action<ConnectionType: PoolManageableConnection> {
+        case lease(ConnectionType, Waiter<ConnectionType>)
+        case create(Waiter<ConnectionType>)
+        case replace(ConnectionType, Waiter<ConnectionType>)
         case closeProvider
-        case park(Connection)
+        case park(ConnectionType)
         case none
-        case fail(Waiter, Error)
-        indirect case closeAnd(Connection, Action)
-        indirect case parkAnd(Connection, Action)
+        case fail(Waiter<ConnectionType>, Error)
+        indirect case closeAnd(ConnectionType, Action<ConnectionType>)
+        indirect case parkAnd(ConnectionType, Action<ConnectionType>)
     }
 
-    struct ConnectionsState {
+    struct ConnectionsState<ConnectionType: PoolManageableConnection> {
```

Comment:
small comment from me: this generic is only needed to we can mock the `Connection` type in order to make sure that `ConnectionState` tests do not depend on `HTTP1ConnectionProvider`.